pkg/controller/nodeconfig/monitor/file.go

```diff
@@ -0,0 +1,86 @@
+package monitor
+
+import (
+	"context"
+
+	gocommon "github.com/harvester/go-common"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ctlv1 "github.com/harvester/node-manager/pkg/generated/controllers/node.harvesterhci.io/v1beta1"
+	"github.com/harvester/node-manager/pkg/utils"
+)
+
+var monitorTargets = []string{utils.SystemdConfigPath}
+var timesyncdConfigPath = utils.SystemdConfigPath + utils.TimesyncdConfigName
+
+type ConfigFileMonitor struct {
+	Context        context.Context
+	MonitorName    string
+	NodeName       string
+	MonitorTargets []string
+
+	NodeConfigCtl ctlv1.NodeConfigController
+}
+
+func NewConfigFileMonitor(ctx context.Context, nodecfg ctlv1.NodeConfigController, nodeName, monitorName string) *ConfigFileMonitor {
+	return &ConfigFileMonitor{
+		Context:        ctx,
+		MonitorName:    monitorName,
+		NodeConfigCtl:  nodecfg,
+		NodeName:       nodeName,
+		MonitorTargets: []string{},
+	}
+}
+
+func (monitor *ConfigFileMonitor) startMonitor() {
+	go func() {
```

Comment:
why do we need to wrap it in another go routine here?

Reply:
Because the monitor is notify-based, it will wait until the notification comes.
We need to use go routine to prevent getting stuck in the main process.